src/test/java/org/folio/dew/batch/bursarfeesfines/service/BursarFilterEvaluatorTest.java

```diff
@@ -0,0 +1,359 @@
+import org.folio.dew.helpers.bursarfeesfines.InvalidBursarExportFilter;
+import org.junit.jupiter.api.Test;
+import org.openapitools.jackson.nullable.JsonNullable;
+
+public class BursarFilterEvaluatorTest {
+
+  @Test
+  void testJsonNullableExportFilter() {
+    AccountWithAncillaryData accountWithAncillaryData = AccountWithAncillaryData
+      .builder()
+      .account(null)
+      .user(null)
+      .item(null)
+      .build();
+
+    BursarExportFilterPass bursarExportFilterPass = new BursarExportFilterPass();
+
+    JsonNullable<BursarExportFilter> jsonNullableFilterPass = JsonNullable.of(
+      bursarExportFilterPass
+    );
+
+    assertThat(
+      BursarFilterEvaluator.evaluate(
+        accountWithAncillaryData,
+        jsonNullableFilterPass
+      ),
+      is(true)
+    );
+
+    assertThat(
+      BursarFilterEvaluator.evaluate(
+        accountWithAncillaryData,
+        JsonNullable.<BursarExportFilter>undefined()
+      ),
+      is(true)
+    );
+  }
+
+  @Test
+  void testFilterAccountByAge() {
+    Account account = new Account();
+
+    AccountWithAncillaryData accountWithAncillaryData = AccountWithAncillaryData
+      .builder()
+      .account(account)
+      .user(null)
+      .item(null)
+      .build();
+
+    BursarExportFilterAge filterAge = new BursarExportFilterAge();
+    filterAge.setNumDays(1);
+
+    assertThat(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterAge),
+      is(true)
+    );
+
+    account.setDateCreated(new Date());
+    accountWithAncillaryData.setAccount(account);
+
+    assertThat(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterAge),
+      is(false)
+    );
+
+    Calendar calendar = Calendar.getInstance();
+    calendar.add(Calendar.DAY_OF_MONTH, -2);
+
+    account.setDateCreated(calendar.getTime());
+    accountWithAncillaryData.setAccount(account);
+
+    assertThat(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterAge),
+      is(true)
+    );
+  }
+
+  @Test
+  void testFilterAccountByAmount() {
+    BursarExportFilterAmount filterAmount = new BursarExportFilterAmount();
+
+    Account account = new Account();
+    account.setAmount(new BigDecimal(50));
+    AccountWithAncillaryData accountWithAncillaryData = AccountWithAncillaryData
+      .builder()
+      .account(account)
+      .user(null)
+      .item(null)
+      .build();
+
+    // test for accounts less than filter value
+    filterAmount.setAmount(6000);
+    filterAmount.setCondition(BursarExportFilterAmount.ConditionEnum.LESS_THAN);
+    assertThat(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterAmount),
+      is(true)
+    );
+
+    // test for accounts less than or equal to filter value
+    filterAmount.setAmount(5000);
+    filterAmount.setCondition(
+      BursarExportFilterAmount.ConditionEnum.LESS_THAN_EQUAL
+    );
+    assertThat(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterAmount),
+      is(true)
```

Comment:
For each condition (less than, greater than, etc), we need to have tests where the filter returns true and returns false.  That's why the four `return`s in the case are yellow: each is missing one of two "branches" (options).

<img width="542" alt="image" src="https://github.com/ualibweb/mod-data-export-worker/assets/8005215/f377bc74-2175-4f42-8a1a-6c7488565cf8">